src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -5131,7 +5155,11 @@ where
         return Err(AdapterError::IntrospectionDisabled { log_names });
     }
 
-    Ok(())
+    Ok(if vars.enable_per_replica_notice() {
+        Some(AdapterNotice::PerReplicaLogRead { log_names })
+    } else {
+        None
+    })
```

Comment:
FYI, you can use `then_some` here to make this a bit shorter:
```rust
vars.enable_per_replica_notice().then_some(AdapterNotice::PerReplicaLogRead { log_names })
```

But it's also fine as it is.